src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -0,0 +1,458 @@
+# Copyright 2025 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.7+
+
+"""TdnfPackageManager for Azure Linux"""
+import datetime
+import os
+import re
+import time
+
+from core.src.package_managers.PackageManager import PackageManager
+from core.src.bootstrap.Constants import Constants
+
+
+class TdnfPackageManager(PackageManager):
+    """Implementation of Azure Linux package management operations"""
+
+    def __init__(self, env_layer, execution_config, composite_logger, telemetry_writer, status_handler):
+        super(TdnfPackageManager, self).__init__(env_layer, execution_config, composite_logger, telemetry_writer, status_handler)
+        # Repo refresh
+        self.cmd_repo_refresh_template = "sudo tdnf clean expire-cache ; sudo tdnf -q list updates"
+
+        # Support to get updates and their dependencies
+        self.tdnf_check = 'sudo tdnf -q list updates'
+        # self.tdnf_check_security_prerequisite = 'sudo tdnf -y install yum-plugin-security'
+        # self.tdnf_check_security = 'sudo tdnf update --assumeno --security'  # TODO: Check with Sam on how tdnf check-update cmd would it return version) or possibly change implementation to this
+        self.single_package_check_versions = 'sudo tdnf list available <PACKAGE-NAME>'
+        self.single_package_check_installed = 'sudo tdnf list installed <PACKAGE-NAME>'
+        self.single_package_upgrade_simulation_cmd = 'sudo tdnf install --assumeno --skip-broken '
+
+        # Install update
+        self.single_package_upgrade_cmd = 'sudo tdnf -y install --skip-broken '
+        self.all_but_excluded_upgrade_cmd = 'sudo tdnf -y update --exclude='
```

Comment:
Is all of this tested on a real machine? Please share example outputs in comments for each call

Reply:
Added test logs in here for different scenarios and the UTs.